src/test/e2e/api/client/feature.optimal304.e2e.test.ts

```diff
@@ -1,220 +1,314 @@
-            type: CHANGE_REQUEST_CREATED,
-            createdBy: testUser.email,
-            createdByUserId: testUser.id,
-            ip: '127.0.0.1',
-            featureName: `ch-on-feature-${apendix}`,
-        });
-
-        shutdownHooks.push(async () => {
-            await app.destroy();
-            await db.destroy();
-        });
+    afterAll(async () => {
+        await app.destroy();
+        await db.destroy();
     });
 
-    test('returns calculated hash', async () => {
+    test('returns calculated hash without if-none-match header (dev env token)', async () => {
         const res = await app.request
             .get('/api/client/features')
+            .set('Authorization', devTokenSecret)
             .expect('Content-Type', /json/)
             .expect(200);
 
-        if (etagVariant.feature_enabled) {
-            expect(res.headers.etag).toBe(`"76d8bb0e:16:${etagVariant.name}"`);
-            expect(res.body.meta.etag).toBe(
-                `"76d8bb0e:16:${etagVariant.name}"`,
-            );
+        if (enabled) {
+            expect(res.headers.etag).toBe(`"76d8bb0e:13:${name}"`);
+            expect(res.body.meta.etag).toBe(`"76d8bb0e:13:${name}"`);
         } else {
-            expect(res.headers.etag).toBe('"76d8bb0e:16"');
-            expect(res.body.meta.etag).toBe('"76d8bb0e:16"');
+            expect(res.headers.etag).toBe('"76d8bb0e:13"');
+            expect(res.body.meta.etag).toBe('"76d8bb0e:13"');
         }
     });
 
-    test(`returns ${etagVariant.feature_enabled ? 200 : 304} for pre-calculated hash${etagVariant.feature_enabled ? ' because hash changed' : ''}`, async () => {
+    test(`returns ${enabled ? 200 : 304} for pre-calculated hash${enabled ? ' because hash changed' : ''} (dev env token)`, async () => {
         const res = await app.request
             .get('/api/client/features')
-            .set('if-none-match', '"76d8bb0e:16"')
-            .expect(etagVariant.feature_enabled ? 200 : 304);
-
-        if (etagVariant.feature_enabled) {
-            expect(res.headers.etag).toBe(`"76d8bb0e:16:${etagVariant.name}"`);
-            expect(res.body.meta.etag).toBe(
-                `"76d8bb0e:16:${etagVariant.name}"`,
-            );
+            .set('Authorization', devTokenSecret)
+            .set('if-none-match', '"76d8bb0e:13"')
+            .expect(enabled ? 200 : 304);
+
+        if (enabled) {
+            expect(res.headers.etag).toBe(`"76d8bb0e:13:${name}"`);
+            expect(res.body.meta.etag).toBe(`"76d8bb0e:13:${name}"`);
         }
     });
 
-    test('returns 200 when content updates and hash does not match anymore', async () => {
-        await app.services.featureToggleService.createFeatureToggle(
-            'default',
-            {
-                name: `featureNew304${apendix}`,
-                description: 'the #1 feature',
-            },
-            TEST_AUDIT_USER,
-        );
+    test('creating a new feature does not modify etag', async () => {
```

Comment:
This is because a feature creation doesn't change the configuration (there's no strategy in it). The previous test was wrong because it did not include the double quotes in the if-none-match header, and therefore it was returning a 200 response.